Replicator/DBAccess.cc

```diff
@@ -121,22 +126,29 @@ namespace litecore { namespace repl {
     }
 
 
+    Retained<C4Document> DBAccess::getDoc(C4Collection* collection,
+                                          slice docID,
+                                          C4DocContentLevel content) const {
+        return useCollection(collection)->getDocument(docID, true, content);
+    }
+
     alloc_slice DBAccess::getDocRemoteAncestor(C4Document *doc) {
         if (_remoteDBID)
             return doc->remoteAncestorRevID(_remoteDBID);
         else
             return {};
     }
     
-    void DBAccess::setDocRemoteAncestor(slice docID, slice revID) {
+    void DBAccess::setDocRemoteAncestor(C4Collection* coll, slice docID, slice revID) {
         if (!_remoteDBID)
             return;
-        logInfo("Updating remote #%u's rev of '%.*s' to %.*s",
-                _remoteDBID, SPLAT(docID), SPLAT(revID));
+        logInfo("Updating remote #%u's rev of '%.*s' to %.*s of collection %.*s.%.*s",
+                _remoteDBID, SPLAT(docID), SPLAT(revID),
+                SPLAT(coll->getSpec().scope), SPLAT(coll->getSpec().name));
         try {
             useLocked([&](C4Database *db) {
                 C4Database::Transaction t(db);
-                Retained<C4Document> doc = db->getDocument(docID, true, kDocGetAll);
+                Retained<C4Document> doc = coll->getDocument(docID, true, kDocGetAll);
```

Comment:
Additional comment for enhancement :

Add `DebugAssert(db.get() == coll->getDatabase());` or use UseCollection() instead of useLocked()